applications/StructuralMechanicsApplication/custom_elements/truss_element_3D2N.cpp

```diff
@@ -408,6 +410,75 @@ namespace Kratos
 		KRATOS_CATCH("")
 	}
 
+	void TrussElement3D2N::CalculateOnIntegrationPoints(
+		const Variable<array_1d<double, 3 > >& rVariable,
+		std::vector< array_1d<double, 3 > >& rOutput,
+		const ProcessInfo& rCurrentProcessInfo) 
+		{
+		if (rVariable == FORCE)
+				{
+					Vector truss_forces = ZeroVector(msDimension);
+					truss_forces[2] = 0.00;
+					truss_forces[1] = 0.00;
+
+					const double internal_strain_gl = this->CalculateGreenLagrangeStrain();

```

Comment:
It seems like this is only needed for the nonlin case
Can you make an if else?
The if for the linear case is already there